django/db/models/fields/__init__.py

```diff
@@ -678,6 +675,12 @@ def contribute_to_class(self, cls, name, virtual_only=False):
             cls._meta.add_field(self, virtual=True)
         else:
             cls._meta.add_field(self)
+        if self.column:
+            # Refuse to override class methods with the descriptor. This leads to a situation
+            # where if you have a classmethod and a field similarly named, then such fields
+            # can't be deferred (we don't have a check for this yet).
+            if not getattr(cls, self.attname, None):
+                setattr(cls, self.attname, DeferredAttribute(self.attname, cls))
```

Comment:
I guess the check needs to be done there as DeferredAttribute might be used as a parent class for a descriptor class with `__set__()` defined.